Source/Eto/Forms/FileDialog.cs

```diff
@@ -11,78 +11,10 @@ namespace Eto.Forms
 	/// <remarks>
 	/// Each filter defines an option for the user to limit the selection of files in the dialog.
 	/// </remarks>
-	public class FileDialogFilter
+	[Obsolete("Since 2.4: Use FileFilter instead")]
+	public class FileDialogFilter : FileFilter
 	{
-		/// <summary>
-		/// Initializes a new instance of the <see cref="Eto.Forms.FileDialogFilter"/> class.
-		/// </summary>
-		public FileDialogFilter()
-		{
-		}
-
-		/// <summary>
-		/// Initializes a new instance of the <see cref="Eto.Forms.FileDialogFilter"/> class.
-		/// </summary>
-		/// <param name="name">Name of the filter to display to the user</param>
-		/// <param name="extensions">Extensions of the files to filter by, including the dot for each</param>
-		public FileDialogFilter(string name, params string[] extensions)
-		{
-			this.Name = name;
-			this.Extensions = extensions;
-		}
```

Comment:
Great progress, looking good!  I think the last thing is we'll still need these constructors to keep compatibility, which should call the base constructors.